uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -129,12 +129,18 @@ define([
     vm._loadCalendar = function () {
       tempContactData = _.clone(vm.managedContacts);
 
-      return $q.all(vm.managedContacts.map(function (contact, index) {
-        return Calendar.get(contact.id, vm.selectedPeriod.id)
-          .then(function (calendar) {
-            tempContactData[index].calendarData = vm._setCalendarProps(vm.managedContacts[index].id, calendar);
+      return Calendar.get(vm.managedContacts.map(function (contact) {
+        return contact.id;
+      }), vm.selectedPeriod.id)
+      .then(function (calendars) {
+        calendars.forEach(function (calendar) {
+          var index = _.findIndex(tempContactData, function (contact) {
```

Comment:
> Can we index tempContactData or vm.managedContacts by id? then performance would increase.
  
honestly i'm not even going to try doing that, this PR needs to be merged today and investigating how to improve how the contacts are stored and indexed and searched in the component would definitely make the PR miss the deadline